Documentation/specs/corehost.md

```diff
@@ -0,0 +1,112 @@
+2. If there is a file in the Application Base with the file name `[AssemblyName].dll`, `[AssemblyName].ni.dll`, `[AssemblyName].exe`, or `[AssemblyName].ni.exe` (in that order), it its full path is used as the Assembly Path
+3. The Assembly Path is resolved out of the Package Caches using the algorithm above (in 'Files from package caches').
+
+A similar process is used to produce a list of Native Library Paths. Native libraries are listed in the `.deps` file (marked with a special `native` type token) and searched in the same way as managed assemblies (Servicing, then app-local, then package caches). The main exception is that the app-local file extensions vary by platform (`.dll` on Windows, `.so` on Linux, `.dylib` on Mac OS X)
+
+Once a the list of assemblies and native libraries is produced, each is deduplicated by taking the first match for an assembly/file name (**open question:** error if two assemblies with the same name?). The managed assemblies are provided in the Trusted Platform Assemblies (TPA) list for the CoreCLR during startup. The folder paths for each native library are deduplicated and provided in the Native Search Paths list for the CoreCLR during startup.
+
+**NOTE**: The CLR may add support for providing a similar structure as the TPA list for native libraries (i.e. a flat list of full file paths).
+
+### Satellite Assemblies
+
+Satellite Assemblies (assemblies containing only embedded resources used in place of the resources provided by an assembly) are stored just like regular assemblies in the TPA list. The CLR will use a path convention to detect which assemblies are culture-neutral and which are culture-specific. The convention will be to look at the last two segments of the path (the file name and the immediate parent directory name). If the parent directory matches an [IETF Language Tag](https://en.wikipedia.org/wiki/IETF_language_tag) (or more specifically, a value usable in the Culture field for a CLR Assembly Name), then the assembly is considered culture-specific (for the culture specified in that folder name).
+
+**NOTE**: The CLR does not do this today, but there will be a work item to track it.
+
+## Runtime Resolution
+
+Runtime resolution is controlled by two environment variables:
+
+* `DOTNET_RUNTIME_SERVICING` -> Global override for runtime
+* `DOTNET_HOME` -> Default location for runtime
+
+The runtime is located by searching the following paths in order, where `APP_BASE` refers to the directory containing the managed application assembly and `LIBCORECLR` refers to the platform-specific name for the CoreCLR library (`libcoreclr.so` on Unix, `libcoreclr.dylib` on Mac OS X, `coreclr.dll` on Windows). The first path that matches is used as the path to load the CoreCLR from.
+
+* `$DOTNET_RUNTIME_SERVICING/runtime/coreclr/LIBCORECLR`
+* `APP_BASE/LIBCORECLR`
+* `$DOTNET_HOME/runtime/coreclr/LIBCORECLR`
+* On Unix: 
+	* `/usr/local/share/dotnet/runtime/coreclr/LIBCORECLR` [1]
```

Comment:
Yes, if `DOTNET_HOME` is not specified, we try each of these until we find `[lib]coreclr.[dll|so|dylib]`